zerver/migrations/0311_put_system_bots_in_all_realms.py

```diff
@@ -0,0 +1,125 @@
+import secrets
+from typing import Any, Optional
+
+import orjson
+from django.conf import settings
+from django.contrib.auth.hashers import make_password
+from django.db import migrations
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.utils.timezone import now as timezone_now
+
+USERPROFILE_DEFAULT_BOT = 1
+USERPROFILE_AVATAR_FROM_GRAVATAR = "G"
+USERPROFILE_TUTORIAL_WAITING = "W"
+RECIPIENT_PERSONAL = 1
+
+
+def generate_api_key() -> str:
+    api_key = ""
+    while len(api_key) < 32:
+        # One iteration suffices 99.4992% of the time.
+        api_key += secrets.token_urlsafe(3 * 9).replace("_", "").replace("-", "")
+    return api_key[:32]
+
+
+def create_bots(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    Realm = apps.get_model("zerver", "Realm")
+    Recipient = apps.get_model("zerver", "Recipient")
+    Subscription = apps.get_model("zerver", "Subscription")
+
+    """
+    Helper functions, copy-pasted from production code and adjusted
+    to work in a migration, for system bot-creating purposes.
+    """
+
+    def create_user(
+        realm: Any,
+        email: str,
+        full_name: str,
+        password: Optional[str] = None,
```

Comment:
I wonder if we should remove all the optional parameters here that we don't invoke in the actual migration.

Reply:
True, cleaned it up.